packages/client/hmi-client/src/components/models/Model.vue

```diff
@@ -646,24 +701,37 @@ const hasNoEmptyKeys = (obj: Record<string, unknown>): boolean => {
 	return Object.keys(nonEmptyKeysObj).length === Object.keys(obj).length;
 };
 
+const createNewModel = async () => {
+	const newModel = {
+		name: newModelName.value,
+		framework: 'Petri Net',
+		description: newDescription.value,
+		content: JSON.stringify(newPetri.value ?? { S: [], T: [], I: [], O: [] })
```

Comment:
This is going to bite us in the *** @mwdchang I know it.

Reply:
Likely won't hurt in the meantime in the case of creating a new empty model.  We should however make sure we can handle the case when it is empty.

Reply:
@YohannParis can you please elaborate? 

Is this because you want to hide the S/D/I/O data structures? Or is it something else?

I don't think it will be the case that the UI will not need to known about STIO stuff - especially with the way the properties and observation functions are going with ensembles that forces you to acknowledge where things are. I think for now it is fine ... probably in the longer term we want to have a `model-interface` to encapsulate some of these yuckiness. Something like

```
content: Petrinet.emptyModel()...
content: ReactionNet.emptyMode()
```

Kind of thing .... and `gene reaction network` is rumoured to be the next modelling framework we will take a crack at.

Reply:
Yes, but I wonder if this should be handle in the back-end, the model object does't have to know in the front-end what is the content of a graph, only the graph-editor. I just think less places where we need to know the format the better.

Reply:
I suspect that will not be case because of the discussions around metadata ... e.g. the metadata are embedded in S and T directly so you have to acknowledge that.

I think we can minimize UI's manipulation of the inner workings ... kind of like a read-only mode unless in the model-editor.

Reply:
I'll let you guys resolve this amongst yourselves.